hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamerMetrics.java

```diff
@@ -157,6 +159,11 @@ public void shutdown() {
     if (metrics != null) {
       metrics.shutdown();
     }
+    // if metadata table is enabled, make sure to shut down the metrics for that table as well
+    if (writeConfig.getMetadataConfig().enabled()) {
+      HoodieWriteConfig metadataWriteConfig = HoodieMetadataWriteUtils.createMetadataWriteConfig(writeConfig, HoodieFailedWritesCleaningPolicy.LAZY);
```

Comment:
Probably better option is:
the metrics within WriteClient should be closed when write client is closed. 

Since we are at this, can you also take care of below?
We also have HoodieMetadataMetrics within HoodieBackedTableMetadataWriter which also need to be closed when the MetadataWriter is closed.